extensions/mssql/src/features.ts

```diff
@@ -86,7 +96,36 @@ export class AccountFeature implements StaticFeature {
 
 		let params: contracts.RequestSecurityTokenResponse = {
 			accountKey: JSON.stringify(account.key),
-			token: securityToken.token
+			token: securityToken.token,
+			expiresOn: securityToken.expiresOn
+		};
+
+		return params;
+	}
+
+	protected async refreshToken(request: contracts.RefreshTokenParams): Promise<contracts.TokenRefreshedParams> {
+
+		// find account
+		const accountList = await azdata.accounts.getAllAccounts();
+		const account = accountList.find(a => a.key.accountId === request.accountId);
+
+		// find tenant
+		const tenant = account.properties.tenants.find(request.tenantId);
+		if (!tenant) {
+			void window.showErrorMessage(localizedConstants.failedToFindTenants);
```

Comment:
If we're going to be showing error messages to the user we have to make it clear where the error is coming from at the very least. Right now we'll just randomly show this error without any indication of what caused it which isn't very useful. 

For now I think it's fine to just not show anything to the user at all. That's what we do currently and ideally we shouldn't be hitting this case much anyways - and then you can follow up with figuring out how to expose this to the user in a clearer manner later on.